cmd/posix.go

```diff
@@ -391,22 +392,33 @@ func (s *posix) getVolDir(volume string) (string, error) {
 func (s *posix) getDiskID() (string, error) {
 	s.RLock()
 	diskID := s.diskID
+	fileInfo := s.formatFileInfo
+	lastCheck := s.formatLastCheck
 	s.RUnlock()
 
+	if fileInfo != nil && diskID != "" && time.Now().Before(lastCheck.Add(time.Second)) {
+		return diskID, nil
+	}
+
+	s.Lock()
```

Comment:
It is probably faster to still use RLock() here and keep using Lock() later as the old code

Reply:
No, then we risk returning inconsistent results. This should only happen once every second, and all 'waiting' will be caught in the first check.